### PR TITLE
feat(pallet-randomness): add benchmarks

### PR DESCRIPTION
**Pull request by jmg-duarte**
### Description

As stated in the title. 

### Checklist

- [x] Have you tested this solution?

### CODE REVIEW

File: pallets/randomness/src/benchmarking.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+#[frame_benchmarking::v2::benchmarks(
+    where T: crate::Config
+)]
+mod benchmarks {
+    use super::*;
+
+    /// The worst case scenario is when the history is full.
+    /// Instead of just adding the randomness value, it requires removing the oldest value too.
+    #[benchmark]
```
Comment (th7nder) [inline pallets/randomness/src/benchmarking.rs:18]: Could you explain how does this work? I mean, I can't see how we are filling up history here.

File: runtime/src/configs/mod.rs [hunk 2]
```diff
@@ -424,4 +424,5 @@ impl pallet_faucet::Config for Runtime {
 /// Config for our randomness pallet
 impl pallet_randomness::Config for Runtime {
     type AuthorVrfGetter = BabeDataGetter<Runtime>;
+    type WeightInfo = ();
```
Comment (th7nder) [inline runtime/src/configs/mod.rs]: What does it do? Why are we setting weight info to nothing?

**Review by th7nder — CHANGES_REQUESTED** [review]

File: runtime/src/configs/mod.rs (hunk 2 above)
Comment (jmg-duarte) [reply runtime/src/configs/mod.rs]: We're not setting it to "nothing", WeightInfo is implemented for unit

File: pallets/randomness/src/benchmarking.rs (hunk 1 above)
Comment (jmg-duarte) [reply pallets/randomness/src/benchmarking.rs:18]: I've fixed it, should be clear now

File: pallets/proofs/src/benchmarking.rs [hunk 3]
```diff
@@ -0,0 +1,63 @@
+#![cfg(feature = "runtime-benchmarks")]
+
+use frame_benchmarking::v2::*;
+use frame_system::RawOrigin;
+
+use super::*;
+#[allow(unused)]
+use crate::Pallet as ProofsPallet;
+
+// NOTE: these benchmarks are cannot be accurate until the extrinsic takes in a bounded key size
```
Comment (th7nder) [inline pallets/proofs/src/benchmarking.rs]: Let's add a todo/link to an existing issue so we don't forget to do this.

File: pallets/weights.rs [hunk 4]
```diff
@@ -0,0 +1,57 @@
+
+//! Autogenerated weights for `pallet_randomness`
```
Comment (th7nder) [inline pallets/weights.rs]: Why this file exists and is in `pallets/weights.rs`, but it is for pallet randomness?

**Review by th7nder — CHANGES_REQUESTED** [review]

File: pallets/weights.rs (hunk 4 above)
Comment (jmg-duarte) [reply pallets/weights.rs]: fixed

File: pallets/proofs/src/benchmarking.rs (hunk 3 above)
Comment (jmg-duarte) [reply pallets/proofs/src/benchmarking.rs]: this needs a bound, otherwise "prod" compilation fails

**Review by th7nder — DISMISSED** [review]

**Review by cernicc — APPROVED** [review]

**Comment by cernicc** [comment]
Good job figuring this out 💪

**Review by th7nder — APPROVED** [review]

**Pull request merged**